File: leetcode/2270-number-of-ways-to-split-array/solution.ts
function waysToSplitArray(nums: number[]): number {
  const prefix: number[] = Array.from({ length: nums.length });
  prefix[0] = nums[0];

  for (let i = 1; i < nums.length; i++) {
    prefix[i] = prefix[i - 1] + nums[i];
  }

  let ans = 0;

  // "i" here is going to be our pivot point
  for (let i = 1; i < nums.length; i++) {
    const leftSum = prefix[i - 1];
    const rightSum = prefix[nums.length - 1] - leftSum;

    if (leftSum >= rightSum) {
      ans++;
    }
  }

  return ans;
}

File: leetcode/2270-number-of-ways-to-split-array/solution.py
def waysToSplitArray(self, nums: List[int]) -> int:
    prefix = [0] * len(nums)

    prefix[0] = nums[0]

    for i in range(1, len(nums)):
        prefix[i] = prefix[i - 1] + nums[i]

    ans = 0

    for i in range(1, len(nums)):
        left_sum = prefix[i - 1]
        right_sum = prefix[len(nums) - 1] - left_sum

        if left_sum >= right_sum:
            ans += 1

    return ans
